### PR TITLE
Adding params key in mongo-state component

### LINKED ISSUE #1311: Mongodb authentication will fail, in case of databasename other than 'admin'
## Expected Behavior

We could set DatabaseName to an name other then 'admin'.
We could set authentication database name.
For mongodb, we should set `--authenticationDatabase`, but there's no such key provided in state.mongo component.  
## Actual Behavior
If databaseName is not 'admin', authentication will fail.
`time="2021-11-18T07:55:17.339768961Z" level=fatal msg="process component chatstore error: error in connecting to mongodb, host: 10.110.100.193:29002 error: connection() error occured during connection handshake: auth error: sasl conversation error: unable to authenticate using mechanism \"SCRAM-SHA-1\": (AuthenticationFailed) Authentication failed." app_id=chatroom instance=chatroom-5b7c88c9c9-sb5f9 scope=dapr.runtime type=log ver=1.4.3`

If database name is not 'damin', authencation will fail, because we don't set authentication database name(--authenticationDatabase). But, there's no such entry in the component manifest.

## Steps to Reproduce the Problem

1. Create a mongodb component, set databaseName 'xxx', while 'xxx' is not 'admin'.
2. Apply that component in k8s.
## Release Note

Add a new key 'auchenticationDatabase' to mongodb component.

RELEASE NOTE:

### PR DESCRIPTION
**Pull request by shivamkm07**
Signed-off-by: shivam <shivamkm07@gmail.com>

Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [ ] [Read the contribution guide](https://docs.dapr.io/contributing/contributing-docs/)
- [ ] Commands include options for Linux, MacOS, and Windows within codetabs
- [ ] New file and folder names are globally unique
- [ ] Page references use shortcodes instead of markdown or URL links
- [ ] Images use HTML style and have alternative text
- [ ] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

Adding params key added in mongo statestore component [ [dapr/components-contrib/pull/216](https://github.com/dapr/components-contrib/pull/216)  ]

## Issue reference

Please reference the issue this PR will close:  https://github.com/dapr/components-contrib/issues/1311

### CODE REVIEW

**Review by mukundansundar — APPROVED** [review]

File: daprdocs/content/en/reference/components-reference/supported-state-stores/setup-mongodb.md [hunk 1]
```diff
@@ -67,6 +69,7 @@ If you wish to use MongoDB as an actor store, append the following to the yaml.
 | writeconcern       | N        | The write concern to use | `"majority"`
 | readconcern        | N        | The read concern to use  | `"majority"`, `"local"`,`"available"`, `"linearizable"`, `"snapshot"`
 | operationTimeout   | N        | The timeout for the operation. Defaults to `"5s"` | `"5s"`
+| params             | N        | Additional parameters to use | `"?authSource=daprStore&ssl=true"`
```
Comment (greenie-msft) [inline daprdocs/content/en/reference/components-reference/supported-state-stores/setup-mongodb.md]: Can you please provide a paragraph below that describes typical use cases that I would need to apply extra params. Also providing a list of the choices that you can apply today.

Comment (shivamkm07) [reply daprdocs/content/en/reference/components-reference/supported-state-stores/setup-mongodb.md]: @greenie-msft Please see the latest changes. Thanks!

**Comment by shivamkm07** [comment]
Also @greenie-msft would this change be backported to older versions (1.5, 1.6 etc.) since `params` key was added much earlier. If needed, I can create a PR against those.

File: daprdocs/content/en/reference/components-reference/supported-state-stores/setup-mongodb.md [hunk 2]
```diff
@@ -67,9 +69,12 @@ If you wish to use MongoDB as an actor store, append the following to the yaml.
 | writeconcern       | N        | The write concern to use | `"majority"`
 | readconcern        | N        | The read concern to use  | `"majority"`, `"local"`,`"available"`, `"linearizable"`, `"snapshot"`
 | operationTimeout   | N        | The timeout for the operation. Defaults to `"5s"` | `"5s"`
+| params             | N<sup>**</sup> | Additional parameters to use | `"?authSource=daprStore&ssl=true"`
 
 > <sup>[*]</sup> The `server` and `host` fields are mutually exclusive. If neither or both are set, Dapr will return an error.
 
+> <sup>[**]</sup> The `params` field accepts a query string that specifies connection specific options as `<name>=<value>` pairs, separated by `"&"` and prefixed with `"?"`. e.g. to use "daprStore" db as authentication database and enabling SSL/TLS in connection, specify params as `"?authSource=daprStore&ssl=true"`. Please see [the mongodb manual](https://docs.mongodb.com/manual/reference/connection-string/#std-label-connections-connection-options) for the list of available options and their use cases.
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-state-stores/setup-mongodb.md]:
```suggestion
> <sup>[**]</sup> The `params` field accepts a query string that specifies connection specific options as `<name>=<value>` pairs, separated by `"&"` and prefixed with `"?"`. e.g. to use "daprStore" db as authentication database and enabling SSL/TLS in connection, specify params as `"?authSource=daprStore&ssl=true"`. See [the mongodb manual](https://docs.mongodb.com/manual/reference/connection-string/#std-label-connections-connection-options) for the list of available options and their use cases.
```

**Comment by msfussell** [comment]
@shivamkm07 - One minor change to remove the Please. No, these are not down merged, we only up merge, from 1.6 to 1.7. We cannot spend the time checking what needs to go down, if the author does not do this. If you would like to push a PR to the previous docs repos that would be fantastic and would help.

**Comment by shivamkm07** [comment]
> @shivamkm07 - One minor change to remove the Please. No, these are not down merged, we only up merge, from 1.6 to 1.7. We cannot spend the time checking what needs to go down, if the author does not do this. If you would like to push a PR to the previous docs repos that would be fantastic and would help.

Thanks for reviewing @msfussell. Change done.
I will create PR against v1.6 and v1.5 as well.

**Pull request merged**